src/SWP/Component/MultiTenancy/Factory/TenantFactory.php

```diff
@@ -13,31 +13,73 @@
  */
 namespace SWP\Component\MultiTenancy\Factory;
 
+use SWP\Component\Common\Generator\GeneratorInterface;
+use SWP\Component\MultiTenancy\Model\TenantInterface;
+use SWP\Component\MultiTenancy\Repository\OrganizationRepositoryInterface;
+use SWP\Component\Storage\Factory\FactoryInterface;
+
 /**
  * Class TenantFactory.
  */
 class TenantFactory implements TenantFactoryInterface
 {
     /**
-     * @var string
+     * @var FactoryInterface
      */
-    protected $className;
+    protected $decoratedFactory;
+
+    /**
+     * @var GeneratorInterface
+     */
+    protected $generator;
+
+    /**
+     * @var OrganizationRepositoryInterface
+     */
+    protected $organizationRepository;
 
     /**
      * TenantFactory constructor.
      *
-     * @param string $className
+     * @param FactoryInterface                $decoratedFactory
+     * @param GeneratorInterface              $generator
+     * @param OrganizationRepositoryInterface $organizationRepository
      */
-    public function __construct($className)
-    {
-        $this->className = $className;
+    public function __construct(
+        FactoryInterface $decoratedFactory,
```

Comment:
no, its Factory from storage component which is being decorated can create you a pure object based on the provided class name in the construct. See storage bundle and storage component documentation. TenantFactory decorates the default one so you can create a tenant with code or with organization.